plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js

```diff
@@ -163,21 +164,64 @@ test.describe( 'Assembler -> Homepage', { tag: '@gutenberg' }, () => {
 		await waitResponse;
 
 		await page.goto( baseURL );
-		// Get all the content between the header and the footer.
-		const homepageHTML = await page
-			.locator(
-				'//header/following-sibling::*[following-sibling::footer]'
-			)
-			.all();
 
-		let index = 0;
-		for ( const element of homepageHTML ) {
-			await expect(
-				await element.getAttribute( 'class' )
-			).toMatchSnapshot( {
-				name: `selected-homepage-blocks-class-frontend-${ index }`,
-			} );
-			index++;
+		// Check if Gutenberg is installed
+		const apiContext = await request.newContext( {
+			baseURL,
+			extraHTTPHeaders: {
+				Authorization: `Basic ${ encodeCredentials(
+					'admin',
+					'password'
+				) }`,
+				cookie: '',
+			},
+		} );
+		const listPluginsResponse = await apiContext.get(
+			`/wp-json/wp/v2/plugins`,
+			{
+				failOnStatusCode: true,
+			}
+		);
+		const pluginsList = await listPluginsResponse.json();
+		const gutenbergPlugin = pluginsList.find(
+			( { textdomain } ) => textdomain === 'gutenberg'
+		);
+
+		// eslint-disable-next-line playwright/no-conditional-in-test
+		if ( gutenbergPlugin ) {
```

Comment:
Something like:

```
name: `${ gutenbergPlugin ? 'gutenberg' : ''}-selected-homepage-blocks-class-frontend-${ index }`
```

Reply:
@adimoldovan That only works if the plugin is installed in the `gutenberg` folder. The `textdomain` filter will find it regardless of what folder it's installed in.

Reply:
I can add a comment about it for better clarification.

Reply:
Now is commented and bit rewritten for better understanding.

Reply:
@adimoldovan That only works if the plugin is installed in the gutenberg folder. The textdomain filter will find it regardless of what folder it's installed in.

I'm not sure what you mean. My suggestion was not about detecting the gutenberg install. I was suggesting another way of using the value of `gutenbergPlugin` which was already set.

Reply:
@veljkho The code that gets all the content between the header and the footer is identical in both `if` and `else` branches.
I was suggesting of using the ternary operator in the snapshot name instead to keep is shorter.

So instead of 
```
if (withGutenbergPlugin) {
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `gutenberg-selected-homepage-blocks-class-frontend-${ index }`,
				}
}
else {
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `selected-homepage-blocks-class-frontend-${ index }`,
				}
}
```

use 

```
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `${ withGutenbergPlugin ? 'gutenberg' : '' }-selected-homepage-blocks-class-frontend-${ index }`,
				}
}
```

Reply:
That's a great improvement @adimoldovan , thanks!